cli/src/stake.rs

```diff
@@ -1486,83 +1509,214 @@ mod tests {
+        let stake_account_pubkey = stake_account_keypair.pubkey();
+        let (stake_authority_keypair_file, mut tmp_file) = make_tmp_file();
+        let stake_authority_keypair = Keypair::new();
+        write_keypair(&stake_authority_keypair, tmp_file.as_file_mut()).unwrap();
 
+        // stake-authorize subcommand
         let stake_account_string = stake_account_pubkey.to_string();
-
-        let (subcommand, authority_flag) = match stake_authorize {
-            StakeAuthorize::Staker => ("stake-authorize-staker", "--stake-authority"),
-            StakeAuthorize::Withdrawer => ("stake-authorize-withdrawer", "--withdraw-authority"),
-        };
-
-        // Test Staker Subcommand
-        let test_authorize = test_commands.clone().get_matches_from(vec![
+        let new_stake_authority = Pubkey::new(&[1u8; 32]);
+        let new_stake_string = new_stake_authority.to_string();
+        let new_withdraw_authority = Pubkey::new(&[2u8; 32]);
+        let new_withdraw_string = new_withdraw_authority.to_string();
+        let test_stake_authorize = test_commands.clone().get_matches_from(vec![
             "test",
-            &subcommand,
+            "stake-authorize",
             &stake_account_string,
+            "--new-stake-authority",
+            &new_stake_string,
+            "--new-withdraw-authority",
+            &new_withdraw_string,
+        ]);
+        assert_eq!(
+            parse_command(&test_stake_authorize, &default_keypair_file, None).unwrap(),
+            CliCommandInfo {
+                command: CliCommand::StakeAuthorize {
+                    stake_account_pubkey,
+                    new_authorizations: vec![
+                        (StakeAuthorize::Staker, new_stake_authority, 0,),
+                        (StakeAuthorize::Withdrawer, new_withdraw_authority, 0,),
+                    ],
+                    sign_only: false,
+                    blockhash_query: BlockhashQuery::All(blockhash_query::Source::Cluster),
+                    nonce_account: None,
+                    nonce_authority: 0,
+                    fee_payer: 0,
+                },
+                signers: vec![read_keypair_file(&default_keypair_file).unwrap().into(),],
+            },
+        );
+        let (withdraw_authority_keypair_file, mut tmp_file) = make_tmp_file();
+        let withdraw_authority_keypair = Keypair::new();
+        write_keypair(&withdraw_authority_keypair, tmp_file.as_file_mut()).unwrap();
+        let test_stake_authorize = test_commands.clone().get_matches_from(vec![
+            "test",
+            "stake-authorize",
             &stake_account_string,
+            "--new-stake-authority",
+            &new_stake_string,
+            "--new-withdraw-authority",
+            &new_withdraw_string,
+            "--stake-authority",
+            &stake_authority_keypair_file,
+            "--withdraw-authority",
+            &withdraw_authority_keypair_file,
```

Comment:
I think it would be worth adding parse_command test cases for:
- `--new-stake-authority` with `--withdraw-authority`
- both `--new-stake-authority` and `--new-withdraw-authority` with only `--withdraw-authority`

Reply:
https://github.com/solana-labs/solana/pull/8976/commits/a9342c52d4de0894e115c31651f48e2d4441df40